### PR TITLE
windows: use upstream k8s pause image

### PR DESCRIPTION
**Pull request by lmm**
## Description

Our Calico for Windows install includes building a pause image using an older version of the pause image Dockerfile. This particular version of the pause image doesn't include wincat.exe which is required for kubectl proxy.

With this change, when a new Windows version comes up we will need to bump our pause image to the latest k8s pause image that supports that version.

`mcr.microsoft.com/oss/kubernetes/pause:3.6` is the latest upstream pause image and appears to support pretty much all windows versions from 1809 onwards (EOL and current).

```
$ crane manifest mcr.microsoft.com/oss/kubernetes/pause:3.6 | grep os\.version
2022/03/04 16:16:44 No matching credentials were found for "mcr.microsoft.com/oss/kubernetes/pause", falling back on anonymous
            "os.version": "10.0.17763.2300"
            "os.version": "10.0.18362.1256"
            "os.version": "10.0.18363.1556"
            "os.version": "10.0.19041.1348"
            "os.version": "10.0.19042.1348"
            "os.version": "10.0.20348.350"
```

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by lmm** [comment]
Pinging @song-jiang 😄

**Comment by caseydavenport** [comment]
@lmm @song-jiang this guy needs a rebase it seems

File: calico/scripts/install-calico-windows.ps1 [hunk 1]
```diff
@@ -49,35 +49,13 @@ function DownloadFiles()
     md $BaseDir\cni\config -ErrorAction Ignore
 
     Write-Host "Downloading Windows Kubernetes scripts"
-    DownloadFile -Url  https://github.com/Microsoft/SDN/raw/master/Kubernetes/windows/hns.psm1 -Destination $BaseDir\hns.psm1
```
Comment (song-jiang) [inline calico/scripts/install-calico-windows.ps1:52]: I think we still need hns.psm1.

File: node/windows-packaging/CalicoWindows/kubernetes/kubelet-service.ps1 [hunk 2]
```diff
@@ -65,7 +65,7 @@ else
     $argList += "--cni-bin-dir=""c:\k\cni"""
     $argList += "--cni-conf-dir=""c:\k\cni\config"""
     $argList += "--network-plugin=cni"
-    $argList += "--pod-infra-container-image=kubeletwin/pause"
+    $argList += "--pod-infra-container-image=mcr.microsoft.com/oss/kubernetes/pause:3.6"
```
Comment (song-jiang) [inline node/windows-packaging/CalicoWindows/kubernetes/kubelet-service.ps1:68]: I suggest we keep using `kubeletwin/pause` in `kubelet-service.ps1`.  We can update our installation script to 
- Download official pause image
- retag it to `kubeletwin/pause`

This approach keeps backward compatibility that pause image name has not been changed and is ready after installation.  It also allows user to install a different pause image if they want.

**Review by song-jiang — CHANGES_REQUESTED** [review]
Two comments.

File: calico/scripts/install-calico-windows.ps1 (hunk 1 above)
Comment (lmm) [reply calico/scripts/install-calico-windows.ps1:52]: Thanks, fixed

File: node/windows-packaging/CalicoWindows/kubernetes/kubelet-service.ps1 (hunk 2 above)
Comment (lmm) [reply node/windows-packaging/CalicoWindows/kubernetes/kubelet-service.ps1]: We talked about this. I've made the change. Thinking about it some more I was a bit unsure about having a local image tag. But I think the main benefit to pre-pulling is figuring out during install time if the pause image exists or not.

File: calico/scripts/install-calico-windows.ps1 [hunk 3]
```diff
@@ -50,35 +50,31 @@ function DownloadFiles()
...
-    PrepareDockerFile
     ipmo C:\k\hns.psm1
-
-    # Prepare POD infra Images
-    c:\k\InstallImages.ps1
-
     InstallK8sBinaries
+
+    # Prepull and tag the pause image for docker
+    if (-not (Get-IsContainerdRunning)) {
```
Comment (song-jiang) [inline calico/scripts/install-calico-windows.ps1:62]: Is pause image used by both docker and containerd? I'm not sure about we need `if not containd` here.

Comment (lmm) [reply calico/scripts/install-calico-windows.ps1:62]: The pause image specified in the kubelet args is only for Docker. For Containerd, this is specified by its own config.

**Review by song-jiang — APPROVED** [review]

**Pull request merged**